numpyro/distributions/continuous.py

```diff
@@ -3265,6 +3265,301 @@ def infer_shapes(
+
+    :param concentration: Degrees of freedom parameter (often denoted :math:`\nu`).
+        Must be greater than `p - 1` where `p` is the dimension of the scale matrix.
+    :param scale_matrix: Positive definite scale matrix :math:`\mathbf{\Psi}`.
+    :param rate_matrix: Inverse of the scale matrix (alternative parameterization).
+    :param scale_tril: Cholesky decomposition of the scale matrix.
+
+    **Properties**
+
+    - **Mean**: :math:`\frac{\mathbf{\Psi}}{\nu - p - 1}` for :math:`\nu > p + 1`
+    - **Mode**: :math:`\frac{\mathbf{\Psi}}{\nu + p + 1}`
+
+    **References**
+
+    [1] https://en.wikipedia.org/wiki/Inverse-Wishart_distribution
+    """
+
+    arg_constraints = {
+        "concentration": constraints.dependent(is_discrete=False, event_dim=0),
+        "scale_matrix": constraints.positive_definite,
+        "rate_matrix": constraints.positive_definite,
+        "scale_tril": constraints.lower_cholesky,
+    }
+    support = constraints.positive_definite
+    reparametrized_params = [
+        "scale_matrix",
+        "rate_matrix",
+        "scale_tril",
+    ]
+
+    def __init__(
+        self,
+        concentration: ArrayLike,
+        scale_matrix: Optional[Array] = None,
+        rate_matrix: Optional[Array] = None,
+        scale_tril: Optional[Array] = None,
+        *,
+        validate_args: Optional[bool] = None,
+    ) -> None:
+        base_dist = InverseWishartCholesky(
+            concentration,
+            scale_matrix,
+            rate_matrix,
+            scale_tril,
+            validate_args=validate_args,
+        )
+        super().__init__(
+            base_dist, CholeskyTransform().inv, validate_args=validate_args
+        )
+
+    @lazy_property
+    def concentration(self):
+        return self.base_dist.concentration
+
+    @lazy_property
+    def scale_matrix(self):
+        return self.base_dist.scale_matrix
+
+    @lazy_property
+    def rate_matrix(self):
+        return self.base_dist.rate_matrix
+
+    @lazy_property
+    def scale_tril(self):
+        return self.base_dist.scale_tril
+
+    @lazy_property
+    def mean(self) -> ArrayLike:
+        # Mean exists only when concentration > p + 1
+        p = self.scale_matrix.shape[-1]
+        return jnp.where(
+            self.concentration[..., None, None] > p + 1,
+            self.scale_matrix / (self.concentration[..., None, None] - p - 1),
+            jnp.full_like(self.scale_matrix, jnp.nan),
+        )
+
+    @lazy_property
+    def mode(self) -> ArrayLike:
+        p = self.scale_matrix.shape[-1]
+        return self.scale_matrix / (self.concentration[..., None, None] + p + 1)
+
+    @lazy_property
+    def variance(self) -> ArrayLike:
+        # Variance of entry (i,j) for nu > p + 3
+        # Var(X_ij) = (Psi_ij^2 + Psi_ii * Psi_jj) / ((nu - p - 1)^2 * (nu - p - 3))
+        p = self.scale_matrix.shape[-1]
+        nu = self.concentration[..., None, None]
```

Comment:
it's better to use `jnp.expand_dims` for `self.concentration` here and above.

Reply:
changed in  in https://github.com/pyro-ppl/numpyro/pull/2103/commits/05127de5c84a959920571b1f3e2cc213a9751ecc